### PR TITLE
restructured and added .gitignore file

### PR DESCRIPTION
**Pull request by DilshodMura**

### CODE REVIEW

File: ProblemSolving/Program.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        //}
+        static void Main(string[] args)
+        {
+            Console.WriteLine("please enter your nubmer...");
+            int input1 = int.Parse(Console.ReadLine());
+            int a = 0;
+            int b = 1;
+            int c;
+            
+            for (int i = 0;; i++)
```
Comment (rustam-umarov) [inline ProblemSolving/Program.cs:21]: `for` обычно используется, когда ты знаешь когда цикл должен остановиться. В твоем случае ты этого не знаешь и выходишь из цикла только когда твое условие попадает в `if statement`, поэтому было бу лучше использовать `while`.

File: ProblemSolving/Program.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                {
+                    Console.WriteLine("Fibonacci");
+                    break;
+                }
+                else if(input1<c)
+                {
+                    Console.WriteLine("не фибоначчи");
+                    break;
+                }
+
```
Comment (rustam-umarov) [inline ProblemSolving/Program.cs:37]: nit: ненужный пробел

File: ProblemSolving/Program.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                    break;
+                }
+                else if(input1<c)
+                {
+                    Console.WriteLine("не фибоначчи");
+                    break;
+                }
+              
+            }
+
```
Comment (rustam-umarov) [inline ProblemSolving/Program.cs:39]: два ненужных пробела

File: ProblemSolving/Program.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                {
+                    Console.WriteLine("не фибоначчи");
+                    break;
+                }
+              
+            }
+            
+
+        }
+
```
Comment (rustam-umarov) [inline ProblemSolving/Program.cs:42]: еще пробел

File: ProblemSolving/Program.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    }
```
Comment (rustam-umarov) [inline ProblemSolving/Program.cs:43]: скобка не на своем уровне

File: ProblemSolving/Program.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            int b = 1;
+            int c;
+            
+            for (int i = 0;; i++)
+            {
+                c = a + b;
+                a = c - a;
+                b = c;
+
+                if (input1 == c)
```
Comment (rustam-umarov) [inline ProblemSolving/Program.cs:27]: постарайся не называть переменную `input1`, дай ей какое-то свойственное название

File: ProblemSolving/Program.cs [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+{
+    class Program
+    {
+
+        //static void Fibonacci(int a)
+        //{
+            
+           
+        //}
+        static void Main(string[] args)
```
Comment (rustam-umarov) [inline ProblemSolving/Program.cs:13]: напиши метод фибоначи в своей отдельном методе и вызывай его из данного `Main` метода

File: ProblemSolving/Program.cs [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            Console.WriteLine("please enter your nubmer...");
```
Comment (rustam-umarov) [inline ProblemSolving/Program.cs:15]: в данном случае ты можешь не считывать инпут, а лучше просто передавай его значение в метод

File: ProblemSolving/Program.cs [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+            int input1 = int.Parse(Console.ReadLine());
+            int a = 0;
```
Comment (rustam-umarov) [inline ProblemSolving/Program.cs:17]: точно также поменяй названия `a, b, c` на что-то, что будет тебе понятно, если тебе придется посмотреть на этот код через год

File: ProblemSolving/Program.cs [hunk 10]
```diff
@@ -0,0 +1,45 @@
+﻿using System;
+
+namespace ProblemSolving
+{
+    class Program
+    {
+
+        //static void Fibonacci(int a)
```
Comment (rustam-umarov) [inline ProblemSolving/Program.cs:8]: коментарии такого рода не приветствуются

File: ProblemSolving.Test/ProblemSolvingTest.cs [hunk 11]
```diff
@@ -0,0 +1,13 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace ProblemSolving.Test
+{
+    [TestClass]
+    public class UnitTest1
+    {
+        [TestMethod]
+        public void TestMethod1()
```
Comment (rustam-umarov) [inline ProblemSolving.Test/ProblemSolvingTest.cs:9]: после того как ты исправишь вещи, про которые я написал, можно приступить к написанию теста, чтобы проверить правильно ли ты решил задачу

**Review by rustam-umarov — CHANGES_REQUESTED** [review]

File: ProblemSolving/Program.cs (hunk 1 above)
Comment (DilshodMura) [reply ProblemSolving/Program.cs:21]: If statement was replaced by while loop
`  for (int i = 0; ; i++)
            {
                result = input + input2;
                input = result - input;
                input2 = result;

                //inserted while loop instead of if else statement
                while ((input1 == result) && (input1 < result))
                {
                    Console.WriteLine("Fibonacci");
                    break;
                }

            }`

Comment (DilshodMura) [reply ProblemSolving/Program.cs:21]: for loop has been replaced on while loop. 
` while (true) 
            {
                result = input + input2;
                input = result - input;
                input2 = result;

                //inserted while loop instead of if else statement
                if (input1 == result)
                {
                    Console.WriteLine("Fibonacci");
                    break;
                }
               else if (input1 < result)
                {
                    Console.WriteLine("Not Fibonacci");
                    break;
                }
                
            }`

File: ProblemSolving/Program.cs (hunk 2 above)
Comment (DilshodMura) [reply ProblemSolving/Program.cs:37]: extra space was taken off

File: ProblemSolving/Program.cs (hunk 7 above)
Comment (DilshodMura) [reply ProblemSolving/Program.cs:13]: Fibonacci method 
`static void Fibonacci()
        {
            var input1 = int.Parse(Console.ReadLine());
            int input = 0;
            int input2 = 1;
            int result;

            while (true) 
            {
                result = input + input2;
                input = result - input;
                input2 = result;

                //inserted while loop instead of if else statement
                if (input1 == result)
                {
                    Console.WriteLine("Fibonacci");
                    break;
                }
               else if (input1 < result)
                {
                    Console.WriteLine("Not Fibonacci");
                    break;
                }
                
            }

        }`

File: ProblemSolving/Program.cs (hunk 9 above)
Comment (DilshodMura) [reply ProblemSolving/Program.cs:17]: a b c was changed to input,input2, and result

File: ProblemSolving/Program.cs (hunk 10 above)
Comment (DilshodMura) [reply ProblemSolving/Program.cs:8]: the comment was changed to new method Fibonacci

File: ProblemSolving.Tests/EasyTasksTests.cs [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+namespace ProblemSolving.Tests
+{
+    [TestClass]
+    public class EasyTasksTests
+    {
+        [TestMethod]
+        public void IsFibonacciHappyPath_13()
+        {
+            var input = 13;
+            var easyTasks = new EasyTasks(); //creating an instance of EasyTasks
```
Comment (rustam-umarov) [inline ProblemSolving.Tests/EasyTasksTests.cs:13]: Looks great! one minor thing to point out - since we're instantiating `EasyTasks` class in multiple places (in IsFibonacciHappyPath_13 and in IsFibonacciFailPath_14), we will need to instantiate it once on class level, so that every method could access it without re-instantiating all the time.

File: README.md [hunk 13]
```diff
@@ -1 +1,15 @@
...
+
+# Project Description
+
+_Project description goes here_
+
+# Rules
+
+- Always make sure you ONLY check in things YOU changed
+- Make sure your solution to the problem is isolated in different method/function
+- Make sure to write at least 3 happy path unit tests and 3 unhappy path unit tests
```
Comment (rustam-umarov) [inline README.md:11]: Please, add 2 more happy path and 2 more fail path unit tests to make sure `IsFibonacci()` method is functioning properly (larger input and negative numbers are highly encouraged).

File: ProblemSolving.Tests/EasyTasksTests.cs [hunk 14]
```diff
@@ -15,13 +16,117 @@ public void IsFibonacciHappyPath_13()
...
+
+        [TestMethod]
+        public void IsSumInputFailPath_1()
+        {
+            var presetValueOne = 10;
+            var presentValueTwo = 5;
+            var easyTasks = new EasyTasks();
+            int failPath = easyTasks.SumOfTwo(presetValueOne, presentValueTwo);
+            Assert.AreNotEqual(25,failPath);
+
```
Comment (rustam-umarov) [inline ProblemSolving.Tests/EasyTasksTests.cs:108]: пробел)

File: ProblemSolving.Tests/EasyTasksTests.cs [hunk 15]
```diff
@@ -15,13 +16,117 @@ public void IsFibonacciHappyPath_13()
...
+        [TestMethod]
+        public void IsSuminputHappyPath_3()
+        {
+            var presetValueOne = 195;
+            var presetValueTwo = 200;
+            var easyTasks = new EasyTasks();
+            int isSuminputHappyPath = easyTasks.SumOfTwo(presetValueOne,presetValueTwo);
+            Assert.AreEqual(395, isSuminputHappyPath);
+        }
+
```
Comment (rustam-umarov) [inline ProblemSolving.Tests/EasyTasksTests.cs:99]: пробел)

File: ProblemSolving.Tests/EasyTasksTests.cs [hunk 16]
```diff
@@ -15,13 +16,117 @@ public void IsFibonacciHappyPath_13()
...
+        [TestMethod]
+        public void IsFibonacciFailPath_10()
+        {
+            var input = 10;
+            var easyTasks = new EasyTasks(); //creating an instance of EasyTasks
+            bool isFibonacci = easyTasks.IsFibonacci(input); //calling method using instance of EasyTasks
+            Assert.IsFalse(isFibonacci);
+
```
Comment (rustam-umarov) [inline ProblemSolving.Tests/EasyTasksTests.cs:64]: пробел)

File: ProblemSolving.Tests/EasyTasksTests.cs [hunk 17]
```diff
@@ -15,13 +16,117 @@ public void IsFibonacciHappyPath_13()
...
+        }
+        [TestMethod]
+        public void IsFibonacciFailPath_12()
+        {
+            var input = 12;
+            var easyTasks = new EasyTasks(); //creating an instance of EasyTasks
+            bool isFibonacci = easyTasks.IsFibonacci(input); //calling method using instance of EasyTasks
+            Assert.IsFalse(isFibonacci);
+
```
Comment (rustam-umarov) [inline ProblemSolving.Tests/EasyTasksTests.cs:54]: пробел)

File: ProblemSolving.Tests/EasyTasksTests.cs [hunk 18]
```diff
@@ -15,13 +16,117 @@ public void IsFibonacciHappyPath_13()
...
+        public void IsFibonacciFailPath_10()
+        {
+            var input = 10;
+            var easyTasks = new EasyTasks(); //creating an instance of EasyTasks
+            bool isFibonacci = easyTasks.IsFibonacci(input); //calling method using instance of EasyTasks
+            Assert.IsFalse(isFibonacci);
+
+        }
+
+        //1. Hppy path
```
Comment (rustam-umarov) [inline ProblemSolving.Tests/EasyTasksTests.cs:67]: не оставляй такие коменты в коде

File: ProblemSolving.Tests/EasyTasksTests.cs [hunk 19]
```diff
@@ -15,13 +16,117 @@ public void IsFibonacciHappyPath_13()
...
+        }
+        [TestMethod]
+        public void IsSumInputFailPath_2()
+        {
+            var presetValueOne = 15;
+            var presentValueTwo = 5;
+            var easyTasks = new EasyTasks();
+            int failPath = easyTasks.SumOfTwo(presetValueOne, presentValueTwo);
+            Assert.AreNotEqual(25, failPath);
+
```
Comment (rustam-umarov) [inline ProblemSolving.Tests/EasyTasksTests.cs:118]: пробел)

File: ProblemSolving.Tests/EasyTasksTests.cs [hunk 20]
```diff
@@ -15,13 +16,117 @@ public void IsFibonacciHappyPath_13()
...
+        }
+        [TestMethod]
+        public void IsSumInputFailPath_3()
+        {
+            var presetValueOne = 13;
+            var presentValueTwo = 6;
+            var easyTasks = new EasyTasks();
+            int failPath = easyTasks.SumOfTwo(presetValueOne, presentValueTwo);
+            Assert.AreNotEqual(25, failPath);
+
```
Comment (rustam-umarov) [inline ProblemSolving.Tests/EasyTasksTests.cs:128]: пробел)

File: ProblemSolving.Tests/EasyTasksTests.cs [hunk 21]
```diff
@@ -15,13 +16,117 @@ public void IsFibonacciHappyPath_13()
...
+        }
+
+    }
```
Comment (rustam-umarov) [inline ProblemSolving.Tests/EasyTasksTests.cs:131]: скобка)

File: ProblemSolving.Tests/EasyTasksTests.cs [hunk 22]
```diff
@@ -15,13 +16,117 @@ public void IsFibonacciHappyPath_13()
...
+        public void IsSumInputFailPath_3()
+        {
+            var presetValueOne = 13;
+            var presentValueTwo = 6;
+            var easyTasks = new EasyTasks();
+            int failPath = easyTasks.SumOfTwo(presetValueOne, presentValueTwo);
+            Assert.AreNotEqual(25, failPath);
+
+        }
+
```
Comment (rustam-umarov) [inline ProblemSolving.Tests/EasyTasksTests.cs:130]: пробел)

File: ProblemSolving.Core/EasyTasks.cs [hunk 23]
```diff
@@ -23,5 +23,12 @@ public bool IsFibonacci(int inputNumber)
                 }
             }
         }
+
+        public int SumOfTwo(int firstInput, int secondInput)
+        {
+            //here goes the solution
```
Comment (rustam-umarov) [inline ProblemSolving.Core/EasyTasks.cs:29]: можно убрать комент и написать чуть проще:
```
public int SumOfTwo(int firstInput, int secondInput)
{
     return firstInput + secondInput;
}
```

File: ProblemSolving.Tests/EasyTasksTests.cs (hunk 12 above)
Comment (rustam-umarov) [reply ProblemSolving.Tests/EasyTasksTests.cs:14]: @DilshodMura посмотрел эти коменты?

**Review by rustam-umarov — APPROVED** [review]
